src/routes/(console)/project-[project]/overview/platforms/wizard/flutter/step1.svelte

```diff
@@ -84,6 +85,11 @@
     };
 
     async function beforeSubmit() {
+        // double-check the hostname value.
+        if (platform === PlatformType.Flutterweb && !isHostnameValid($createPlatform.hostname)) {
+            throw new Error('Please enter a valid hostname');
+        }
+
```

Comment:
I thought we typically pass a regex to the input for these kinds of validations. Why can't we do that here?

Reply:
1. extracted function can be helpful for future
2. We could use regex pattern, but this way also helps in disabling the next button

Reply:
Re-usable is good.

I still have some concerns:
1. This pattern is different compared to other places in the Console.
2. There is little feedback to the user about what's wrong.

For example, here:

<img width="840" alt="image" src="https://github.com/user-attachments/assets/1fd7005b-5c0e-49ce-b78c-2ddfb2aae6e3" />

Yes, the button is disabled, but I don't know why. Compared to other places like create document: 

<img width="1173" alt="image" src="https://github.com/user-attachments/assets/89f6d623-eac4-4fea-9ae6-516b5bf2e21e" />

As you can see for the `asdf` attribute, there's more feedback to the developer about what's wrong.

Reply:
Makes sense. I can add a regex with the existing logic and that should give a clearer idea on why its disabled. Thanks for the screenshots 👍

Reply:
@stnguyen90 made a few changes to overall logic. It doesn't seem possible to directly manage the warning helper via the `InputText` due to regex complexity as Svelte breaks it. Let me know if the newer changes are OK (backwards compatible).

Reply:
Ya, we've dealt the regex problem in the past. It seems there's a special type of regex that's used in HTML inputs

Reply:
What if we update the inputText input to expose a pattern and patternError like inputOTP:

https://github.com/appwrite/console/blob/e5ab42659fdc5324e2fc3734481c8c48f2c7b985/src/lib/elements/forms/inputOTP.svelte#L12-L13

As for the pattern, I think this should work:

```
const pattern = String.raw`(\*\.)?(?!-)[A-Za-z0-9\-]+([\-\.]{1}[a-z0-9]+)*\.[A-Za-z]{2,18}`;
```

We might also want to create a new input because we're going to also need it on the update platform pages such as:

https://github.com/appwrite/console/blob/e5ab42659fdc5324e2fc3734481c8c48f2c7b985/src/routes/(console)/project-%5Bproject%5D/overview/platforms/%5Bplatform%5D/web.svelte#L48-L53

Reply:
That regex wouldn't have allowed `*`, updated with latest changes.  No changes to wizard required now.

Reply:
Good catch!